src/features/atomics.md

```diff
@@ -0,0 +1,202 @@
+- T1 is holding the lock and T2 is waiting for it with `Atomics.wait`. T1 releases the lock - T2 wakes up (the `Atomics.wait` call returns) and tries to do `Atomics.compareExchange` to acquire the lock, but another thread T3 was faster and got the lock already. So the call to `Atomics.compareExchange` fails to get the lock, and T2 calls `Atomics.wait` again, blocking until T3 releases the lock.
+
+Because of the latter corner case, the mutex isn’t "fair". It’s possible that T2 has been waiting for the lock to be released, but T3 comes and gets it immediately. A more realistic lock implementation may use several states to differentiate between "locked" and "locked with contention".
+
+## Async lock
+
+The non-blocking `executeLocked` method is callable from the main thread, unlike the blocking `lock` method. It gets a callback function as its only parameter and schedules the callback to be executed once it has successfully acquired the lock.
+
+```js
+executeLocked(f) {
+  const self = this;
+
+  function tryGetLock() {
+    while (true) {
+      const oldValue = Atomics.compareExchange(self.i32a, AsyncLock.INDEX,
+                          /* old value >>> */  AsyncLock.UNLOCKED,
+                          /* new value >>> */  AsyncLock.LOCKED);
+      if (oldValue == AsyncLock.UNLOCKED) {
+        f();
+        self.unlock();
+        return;
+      }
+      const result = Atomics.waitAsync(self.i32a, AsyncLock.INDEX,
+                                       AsyncLock.LOCKED
+                                   /*  ^ expected value at start */);
+      if (result.value == "not-equal") {
+        continue;
+      }
+      result.value.then(tryGetLock);
+      return;
+    }
+  }
+
+  tryGetLock();
+}
+```
+
+The inner function `tryGetLock` tries to first get the lock with `Atomics.compareExchange`, as before. If that successfully changes the lock state, it can execute the callback, unlock the lock, and return.
+
+If `Atomics.compareExchange` fails to get the lock, we need to try again when the lock is probably free. We can’t block and wait for the lock to become free - instead, we schedule the new try using `Atomics.waitAsync` and the Promise it returns.
+
+If we successfully started `Atomics.waitAsync`, the returned Promise will resolve when the lock-holding thread does `Atomics.notify`. Then the thread that was waiting for the lock will try to get the lock again, like before.
+
+The same corner cases (the lock getting released between the `Atomics.compareExchange` call and the `Atomics.waitAsync` call, as well as the lock getting acquired again between the Promise resolving and the `Atomics.compareExchange` call) are possible in the asynchronous version too, so the code has to handle them in a robust way.
+
+## Conclusion
```

Comment:
We usually have a feature support table in the end of the feature explainer posts, let's add one here too.

Reply:
done